pdf/pdflinkarea.cpp

```diff
@@ -66,23 +73,40 @@ void PDFLinkArea::mousePressEvent(QMouseEvent *event)
 {
     d->clickInProgress = true;
     d->clickLocation = event->pos();
+    d->pressTimer.start();
 }
 
-void PDFLinkArea::mouseReleaseEvent(QMouseEvent *event)
+void PDFLinkArea::mouseMoveEvent(QMouseEvent *event)
 {
+    // Don't activate anything if the finger has moved too far
+    QRect rect((d->clickLocation - QPointF(d->wiggleFactor, d->wiggleFactor)).toPoint(),
+               QSize(d->wiggleFactor * 2, d->wiggleFactor * 2));
+    if (!rect.contains(event->pos()))
+        return;
     d->clickInProgress = false;
```

Comment:
Well, in fact, clickInProgress has a different meaning than the timer. Both the timer and clickInProgress are triggered on press event. Then a motion event out of the wiggle should cancel both. Then the timer trigger and later click event on released is cancelled by clickInProgress to false.

But anyway, you are right there was a bug in the PR that makes moving a bit inside the wiggle cancel later click on release because the clickInProgress = false was not in the if. It's corrected.